jcommune-view/jcommune-web-view/src/main/webapp/resources/javascript/app/topicDraft.js

```diff
@@ -0,0 +1,559 @@
+        MAX_ITEMS_NUMBER = 50;
+
+    var TITLE_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_TITLE_LENGTH).replace('{max}', MAX_TITLE_LENGTH),
+
+        CONTENT_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_CONTENT_LENGTH).replace('{max}', MAX_CONTENT_LENGTH),
+
+        POLL_TITLE_ERROR_MESSAGE = $labelMessageSizeValidation
+            .replace('{min}', MIN_POLL_TITLE_LENGTH).replace('{max}', MAX_POLL_TITLE_LENGTH),
+
+        POLL_ITEMS_VALUE_ERROR_MESSAGE = $pollItemsSize
+            .replace('{min}', MIN_ITEMS_NUMBER).replace('{max}', MAX_ITEMS_NUMBER),
+
+        POLL_ITEM_LENGTH_ERROR = $pollItemLength
+            .replace('{min}', MIN_POLL_ITEM_LENGTH).replace('{max}', MAX_POLL_ITEM_LENGTH);
+
+    /**
+     * Class for access to topic draft API
+     *
+     * @constructor
+     */
+    function TopicDraftApi() {
+        this._saveLock = false;
+        this._removeLock = false;
+    }
+
+    TopicDraftApi.prototype = {
+        /**
+         * Saves or update draft.
+         *
+         * @param data
+         * @param async
+         * @returns {$.Deferred}
+         */
+        save: function (data, async) {
+            var self = this;
+
+            async = async || true;
+
+            if (self._saveLock) {
+                return $.Deferred(function (deferred) {
+                    deferred.fail();
+                });
+            }
+
+            self._saveLock = true;
+
+            return $.ajax({
+                url: baseUrl + '/topics/draft',
+                type: 'POST',
+                async: async,
+                data: JSON.stringify(data),
+                contentType: "application/json"
+            }).always(function () {
+                self._saveLock = false;
+            });
+        },
+
+        /**
+         * Deletes draft.
+         *
+         * @returns {$.Deferred}
+         */
+        remove: function () {
+            var self = this;
+
+            if (self._removeLock) {
+                return $.Deferred(function (deferred) {
+                    deferred.fail();
+                });
+            }
+
+            self._removeLock = true;
+
+            return $.ajax({
+                type: 'DELETE',
+                url: baseUrl + '/topics/draft'
+            }).always(function() {
+                self._removeLock = false;
+            });
+        }
+    };
+
+    /**
+     * Helper for calling functions at specified interval
+     *
+     * @param {Function} fn the called function
+     * @param {Number} intervalTime the interval in milliseconds
+     * @constructor
+     */
+    function Timer(fn, intervalTime) {
+        this._fn = fn;
+        this._intervalTime = intervalTime;
+        this._intervalID = null;
+    }
+
+    Timer.prototype = {
+        start: function () {
+            if (this._intervalID === null) {
+                this._intervalID = setInterval(this._fn, this._intervalTime);
```

Comment:
Mm.. Are you sure we want to use `setInterval()` instead of `setTimeout()`? To me `setTimeout()` should be a default choice since it's more "polite" to the browser and we won't risk with multiple saving processes going right one after each other.